src/Psy/Command/SandboxCommand.php

```diff
@@ -0,0 +1,315 @@
+<?php
+
+namespace Psy\Command;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class SandboxCommand
+ * @package Psy\Command
+ */
+class SandboxCommand extends Command
+{
+    const SANDBOX_FOLDER = 'psysh_sandbox';
+
+    const ACTION_ADD = 'add';
+    const ACTION_CREATE = 'create';
+    const ACTION_LIST = 'list';
+    const ACTION_WHICH = 'which';
+    const ACTION_USE = 'use';
+    const ACTION_SWITCH = 'switch';
+    const ACTION_DELETE = 'delete';
+    const ACTION_CLEAR = 'clear';
+    const ACTION_EXIT = 'exit';
+    const ACTION_RESET = 'reset';
+    const ACTION_HELP = 'help';
+
+    /**
+     * @var String
+     */
+    protected $tempDir;
+
+    /**
+     * @var array
+     */
+    protected $sandboxes = array();
+
+    /**
+     * @var String
+     */
+    protected $restoreFolder;
+
+    /**
+     * @var string
+     */
+    protected $current;
+
+    /**
+     * @var InputInterface
+     */
+    protected $input;
+
+    /**
+     * @var OutputInterface
+     */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('sandbox')
+            ->setDefinition(array(
+                new InputArgument(
+                    'action',
+                    InputArgument::OPTIONAL,
+                    'Action to perform : add|create, list, which, use|switch, clear|delete, exit|reset'
+                ),
+                new InputArgument('args', InputArgument::IS_ARRAY | InputArgument::OPTIONAL, ''),
+                new InputOption(
+                    'grep',
+                    null,
+                    InputOption::VALUE_OPTIONAL,
+                    'grep parameter'
+                ),
+            ))
+            ->setDescription('Manages sandboxed environments.')
+            ->setHelp(
+                <<<HELP
+sandbox add|create [name] creates a encapsulated sandbox with an optional name, if not provided will autogenerate a name.
+sandbox list Will show a list of the currently generated sandboxes
+sandbox which Will show the current sandbox
+sandbox use|switch name Switches to sandbox by name,
+sandbox clear|delete [name] Removes a sandbox by name, or the current if name is not provided.
+sandbox exit|reset Erases all sandboxes and sets the normal state.
+HELP
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->input = $input;
+        $this->output = $output;
+
+        // extract real command name
+        $tokens = preg_split('{\s+}', $input->__toString());
```

Comment:
same thing here, why do you bother to parse the input when it should already been parsed by the Application ?